### PR TITLE
Allow pre-release versions to pass version constraints

### LINKED ISSUE #3402: Pre-releases break version constraints
I should probably also mention that the beta build broke the TG version check we had:
```
% terragrunt apply
18:24:40.515 ERROR  The currently installed version of Terragrunt (0.67.5-beta2024091201) is not compatible with the version constraint requiring (>= 0.55.2).
18:24:40.515 ERROR  Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```
It was easy to fix by commenting out the version constraint, but if you start to do more beta builds you'll probably want to figure out why the check is failing.

_Originally posted by @jyoder-deloitte in https://github.com/gruntwork-io/terragrunt/pull/3400#issuecomment-2347832016_

### PR DESCRIPTION
**Pull request by yhakbar**
## Description

Fixes #3402.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Updated `version_constraint` logic to ignore pre-release information.

### CODE REVIEW

File: test/integration_test.go [hunk 1]
```diff
@@ -2409,37 +2412,45 @@ func TestTerragruntVersionConstraints(t *testing.T) {
...
+			false,
+		},
 	}
 
-	for _, tt := range tc {
-		tt := tt
-
+	for _, tt := range tc { //nolint:paralleltest
 		t.Run(tt.name, func(t *testing.T) {
-			t.Parallel()
```
Comment (levkohimins) [inline test/integration_test.go:2441]: Why can't these subtests be run in parallel? Is this because of some global variable?

**Review by levkohimins — CHANGES_REQUESTED** [review]

File: test/integration_test.go (hunk 1 above)
Comment (yhakbar) [reply test/integration_test.go:2441]: Ya, see [this](https://github.com/gruntwork-io/terragrunt/pull/3405/files#diff-8118e44ac15fb5ab2e2ece2ab52fd23062a10b5cc9c55ee5b3620a1242cff27cR2396-R2398). The `runTerragruntVersionCommand` modifies the global `version.Version`.

Comment (levkohimins) [reply test/integration_test.go:2441]: Ah, Understood.

**Review by levkohimins — APPROVED** [review]
LGTM!

**Review by denis256 — APPROVED** [review]

**Pull request merged**